codec/types/interface_registry.go

```diff
@@ -43,6 +46,16 @@ type InterfaceRegistry interface {
 	//  registry.RegisterImplementations((*sdk.Msg)(nil), &MsgSend{}, &MsgMultiSend{})
 	RegisterImplementations(iface interface{}, impls ...proto.Message)
 
+	// RegisterServiceRequestType allows a protobuf service method request type to be
+	// registered as the google.protobuf.Any value type for type URLs corresponding
+	// to the fully qualified method name. iface should be an interface as type
+	// as in RegisterInterface and RegisterImplementations.
+	//
+	// This will allow us to pack service methods in Any's using the full method name
+	// as the type URL and the request body as the value, and allow us to unpack
+	// such packed methods using the normal UnpackAny method for the interface iface.
+	RegisterServiceRequestType(iface interface{}, methodName string, request proto.Message)
```

Comment:
Maybe this should just be called something like `RegisterCustomTypeURL` instead because that's really what it does under the hood. It doesn't actually care if that type URL is a method name or whatever. We're just mapping method names to request types for the purpose of using with `AnyResolver` which seems a reasonable customization. What do you think @amaurymartiny ?

Reply:
Yes, from the POV of the interface registry, `RegisterCustomTypeURL` makes more sense, the registry doesn't need to know about service msgs.